### DISCUSSION TITLE
Best way to pull list of components and corresponding values off of an entity?

### DISCUSSION THREAD

**Discussion by DanielMitchellDev** [Doraku/DefaultEcs#126] [Q&A]
Hey,

Been stuck on this one for a minute. Is there a way to get a list of  ALL components and the values stored in those components off of an entity (for example, in some sort of Dictionary form) so that I can convert them into a json project I'm working on? 

so far for each entity all I can find is the .get<componenttype>() method which requires you know what components are on the entity and you're only able to get 1 component at a time.

**Comment by Doraku** [comment]
Look into https://github.com/Doraku/DefaultEcs/blob/master/documentation/api/Entity_ReadAllComponents(IComponentReader).md
You have to provide your own implementation of IComponentReader which will be called back with all the component of the entity (both supplied serializer in DefaultEcs use this).

The debug view for entity also do the same thing so you can see an entity components in the debuger if you want to see some code example https://github.com/Doraku/DefaultEcs/blob/master/source/DefaultEcs/Technical/Debug/EntityDebugView.cs

**Reply by DanielMitchellDev** [reply]
Thanks, I think this took me in the right direction!
